examples/pennylane_run_doubly_stochastic.py

```diff
@@ -0,0 +1,312 @@
+
+.. math:: \langle A_i \rangle = \langle 0 | U(\theta)^\dagger A_i U(\theta) | 0\rangle
+
+for a set of observables :math:`\{A_i\}`, and loss function
+
+.. math:: \mathcal{L}(\theta, \langle A_1 \rangle, \dots, \langle A_M \rangle).
+
+While the expectation values can be calculated analytically in classical simulations,
+on quantum hardware we are limited to *sampling* from the expectation values; as the
+number of samples (or shots) increase, we converge on the analytic expectation value, but can
+never recover the exact expression. Furthermore, the parameter-shift rule
+(`Schuld et al., 2018 <https://arxiv.org/abs/1811.11184>`__) allows for analytic
+quantum gradients to be computed from a linear combination of the variational circuits'
+expectation values.
+
+Putting these two results together, `Sweke et al. (2019) <https://arxiv.org/abs/1910.01155>`__
+show that samples of the expectation value fed into the parameter-shift rule provide
+unbiased estimators of the quantum gradient---resulting in a form of stochastic gradient descent.
+Moreover, they show that convergence of the stochastic gradient descent is guaranteed
+to be on par with standard gradient descent, even in the case where the number
+of shots is 1!
+
+.. note::
+
+    It is worth noting that the smaller the number of shots used, the larger the
+    variance in the estimated expectation value. As a result, it may take
+    more optimization steps for convergence than using a larger number of shots,
+    or an exact value.
+
+    At the same time, a reduced number of shots may significantly reduce the
+    wall time of each optimization step, leading to a reduction in the overall
+    optimization time.
+
+Let's consider a simple example in PennyLane, comparing analytic gradient
+descent (with exact expectation values) to stochastic gradient descent
+using a finite number of shots.
+
+"""
+
+##############################################################################
+# A single-shot stochastic gradient descent
+# -----------------------------------------
+#
+# Consider the Hamiltonian
+#
+# .. math::
+#
+#     H = \begin{bmatrix}
+#           8 & 4 & 0 & -6\\
+#           4 & 0 & 4 & 0\\
+#           0 & 4 & 8 & 0\\
+#           -6 & 0 & 0 & 0
+#         \end{bmatrix}.
+#
+# We can solve for the ground state energy using
+# the variational quantum eigensolver (VQE) algorithm.
+#
+# Let's use the ``default.qubit`` simulator for both the analytic gradient,
+# as well as the estimated gradient using number of shots :math:`N\in\{1, 100\}`.
+
+import pennylane as qml
+from pennylane import numpy as np
+
+np.random.seed(2)
+
+from pennylane import expval
+from pennylane.init import strong_ent_layers_uniform
+from pennylane.templates.layers import StronglyEntanglingLayers
+
+num_layers = 2
+num_wires = 2
+eta = 0.01
+steps = 200
+
+dev_GD = qml.device("default.qubit", wires=num_wires, analytic=True)
+dev_SGD = qml.device("default.qubit", wires=num_wires, analytic=False)

```

Comment:
I would prefer you call them `dev_analytic` and `dev_stochastic` or something along these lines. It would then make sense to use `dev_analytic` to track the real cost function.